pennylane/templates/subroutines/multiplier.py

```diff
@@ -237,3 +253,74 @@ def compute_decomposition(
             op_list.append(qml.adjoint(op))
 
         return op_list
+
+
+def _multiplier_decomposition_resources(
+    num_x_wires,
+    mod,
+    num_work_wires,
+) -> dict:
+    if mod != 2**num_x_wires:
+        num_wires_aux = num_work_wires - 1
+    else:
+        num_wires_aux = num_x_wires
+
+    resources = {
+        resource_rep(qml.QFT, num_wires=num_wires_aux): 2,
+        resource_rep(
+            qml.ControlledSequence,
+            base=qml.PhaseAdder,
+            base_params={"num_x_wires": num_wires_aux, "mod": mod},
+            num_control_wires=num_x_wires,
+        ): 1,
+        adjoint_resource_rep(
+            qml.ControlledSequence,
+            {
+                "base": qml.PhaseAdder,
+                "base_params": {"num_x_wires": num_wires_aux, "mod": mod},
+                "num_control_wires": num_x_wires,
+            },
+        ): 1,
+        adjoint_resource_rep(qml.QFT, {"num_wires": num_wires_aux}): 2,
+    }
+
+    for _ in range(num_x_wires):
+        rep = resource_rep(qml.SWAP)
+        if rep in resources:
+            resources[rep] += 1
+        else:
+            resources[rep] = 1
```

Comment:
You can make `resources` a `Counter`, which takes care of this automatically. See https://docs.python.org/3/library/collections.html#collections.Counter